### ISSUE TITLE
Minor styling issue

### ISSUE THREAD

**Issue by seanevd** [thysultan/stylis.js#39]
I'm using stylis through styled-components and you already fixed an issue I had a few months ago (https://github.com/styled-components/styled-components/issues/530) but I found another one when I wrote this
```
.item {
  .wrapper button& {
    /* styles */
  }
}
```
It should out put `.wrapper button.item {}` but it wasn't working for me. It works as expected on styled-components 1.X.

**Comment by thysultan** [comment]
What does 1.X output exactly? shouldn't it be `.wrapper button.ID .item` since `item{}` is actually a child of the components namespace `ID`?

**Comment by seanevd** [comment]
Hmm... possibly. I don't know what `ID` would be. But I think the important part would be zero spaces between the `button` element and the `.item` class, which is what I was trying to achieve.

Looking back at the output it is the same as what I posted before.

**Comment by thysultan** [comment]
>Looking back at the output it is the same as what I posted before.

Is this 1.X output? Do know what 1.X produces? It should have a namespace hash that styled-components generates which is what `ID` in my comment was a placeholder for.

**Comment by seanevd** [comment]
Yeah, I've changed class names. I guess if I were to show you how I'm doing it with styled components it would look more like this:
```
const Item = styled.div`
  .wrapper button& {
    /* styles */
  }
`
```
And the output should look like this `.wrapper button.iwwPzy`

**Comment by thysultan** [comment]
Thanks, patched in 3.1.9
